cns/restserver/util.go

```diff
@@ -40,6 +40,32 @@ func (service *HTTPRestService) setNetworkInfo(networkName string, networkInfo *
 	service.state.Networks[networkName] = networkInfo
 }
 
+func (service *HTTPRestService) SetPnpIDMacaddressMapping() error {
+	service.Lock()
+	defer service.Unlock()
```

Comment:
I would strongly recommend not using this. Embedding a `sync.RWMutex` is almost never a good idea unless you're implementing another kind of synchronization primitive (`sync.Cond` is a good example). Consequently, its embedding in `*HTTPRestService` is probably a mistake. Plus, it's not clear exactly what else is using this, so this runs the risk of deadlocking (maybe not now, but perhaps when future changes are made).

Instead, if you want a synchronized cache, I'd recommend creating a type to do exactly that with its own mutex to manage *only* its critical sections.

Reply:
This gets called in two places(main during start up and utils), I added this to ensure that cache doesn't get overwritten. How does this result in deadlock?

Reply:
Not necessarily, Overwriting is fine since values will remain constant. I will remove that section.